Library/Homebrew/service.rb

```diff
@@ -351,6 +353,18 @@ def macos_legacy_timers(value = T.unsafe(nil))
       end
     end
 
+    sig { params(value: Integer).returns(T.nilable(Integer)) }
+    def nice(value = T.unsafe(nil))
+      if value.nil?
+        @nice
+      else
+        raise TypeError, "Service#nice value should be in #{NICE_RANGE}" unless NICE_RANGE.cover?(value)
+
+        @require_root = true if value.negative?
```

Comment:
I think this should be explicit rather than implicit. I'd suggest throwing an error later if it's set to a negative value but `require_root` isn't set.